File: examples/cpp/demo_usage.cpp
#include "enhancex/enhancex.hpp"
#include <iostream>

int main() {
    std::cout << "=== EnhanceX C++20 API Demonstration ===\n";

    auto& gpu = enhancex::GPUManager::getInstance();
    std::cout << "Active Device: " << gpu.getDeviceName() << "\n";

    enhancex::ImageEnhancer enhancer;
    cv::Mat input = cv::Mat::zeros(400, 400, CV_8UC3);
    cv::putText(input, "C++ EnhanceX", cv::Point(50, 200), cv::FONT_HERSHEY_SIMPLEX, 1.0, cv::Scalar(255, 255, 255), 2);

    cv::Mat sharpened = enhancer.sharpen(input, 1.5f);
    cv::Mat claheResult = enhancer.applyCLAHE(sharpened, 2.0);

    std::cout << "Successfully enhanced C++ frame: " << claheResult.cols << "x" << claheResult.rows << "\n";
    return 0;
}

File: examples/python/demo_usage.py
"""
EnhanceX Python API Demonstration Script
"""
import numpy as np
import cv2
from enhancex import VideoEnhancer, ImageEnhancer, Stabilizer


def main():
    print("=== EnhanceX Python API Demo ===")

    # 1. Image Enhancement
    img_enhancer = ImageEnhancer()
    dummy_img = np.zeros((400, 400, 3), dtype=np.uint8)
    cv2.putText(dummy_img, "EnhanceX Demo", (50, 200), cv2.FONT_HERSHEY_SIMPLEX, 1.0, (255, 255, 255), 2)

    enhanced_img = img_enhancer.enhance(
        dummy_img,
        sharpen=1.5,
        clahe=True,
        white_balance=True,
        denoise=5.0
    )
    print(f"Enhanced Image Shape: {enhanced_img.shape}")

    # 2. Video Enhancer API
    enhancer = VideoEnhancer()
    print("VideoEnhancer API Initialized successfully.")


if __name__ == "__main__":
    main()
